### PR TITLE
Added a new class that deals with pagination for the Donations and Re…

### PR DESCRIPTION
**Pull request by adriangabrieloros**
…port Items.

Also modified the detail html accordingly.

### Requirements for making a pull request

Thank you for contributing to our project! 

Please fill out the template below to help the project maintainers review it as fast as possible and include your contribution to the project.

### What does it fix?

Closes 

Please mention the main changes this PR brings.

### How has it been tested?

Please describe the tests that you ran to verify your changes.

### CODE REVIEW

File: ro_help/hub/views.py [hunk 1]
```diff
@@ -75,7 +75,42 @@ def get_context_data(self, **kwargs):
...
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        ngo = kwargs.get("ngo", context.get("ngo"))
+        if not ngo:
+            return context
+        page = self.request.GET.get("page")
+
+        # Donations paginator.
+        ngo_donations = ngo.get_funders()
+        donations_paginator = paginator.Paginator(ngo_donations.order_by("id"), self.paginated_by)
```
Comment (vtemian) [inline ro_help/hub/views.py]: Let's order after `created`, desc.

File: ro_help/hub/views.py [hunk 2]
```diff
@@ -75,7 +75,42 @@ def get_context_data(self, **kwargs):
...
+    """
+
+    paginated_by = 10
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        ngo = kwargs.get("ngo", context.get("ngo"))
+        if not ngo:
+            return context
+        page = self.request.GET.get("page")
```
Comment (vtemian) [inline ro_help/hub/views.py]: Let's add prefixes for pagination types: `donation_page`.

File: ro_help/hub/views.py [hunk 3]
```diff
@@ -75,7 +75,42 @@ def get_context_data(self, **kwargs):
...
+
+        # Donations paginator.
+        ngo_donations = ngo.get_funders()
+        donations_paginator = paginator.Paginator(ngo_donations.order_by("id"), self.paginated_by)
+        try:
+            donations_page_obj = donations_paginator.page(page)
+        except (paginator.PageNotAnInteger, paginator.EmptyPage):
+            donations_page_obj = donations_paginator.page(1)
+        context[f"donations_page_obj"] = donations_page_obj
```
Comment (vtemian) [inline ro_help/hub/views.py]: No need for `f"`

File: ro_help/hub/views.py [hunk 4]
```diff
@@ -75,7 +75,42 @@ def get_context_data(self, **kwargs):
...
+
+        # Report items paginator.
+        ngo_report_items = ngo.report_items.all()
+        report_items_paginator = paginator.Paginator(ngo_report_items.order_by("id"), self.paginated_by)
+        try:
+            report_items_page_obj = report_items_paginator.page(page)
+        except (paginator.PageNotAnInteger, paginator.EmptyPage):
+            report_items_page_obj = report_items_paginator.page(1)
+        context[f"report_items_page_obj"] = report_items_page_obj
```
Comment (vtemian) [inline ro_help/hub/views.py]: Same as above.

File: ro_help/hub/templates/ngo/detail.html [hunk 5]
```diff
@@ -240,6 +245,11 @@ <h2 class="title">
 
         <div id="reporting" class="is-collapsible">
             <div class="card-content donation-card">
+
+                <div class="container">
+                    {% include "shared/pagination.html" with page_obj=report_items_page_obj %}
```
Comment (vtemian) [inline ro_help/hub/templates/ngo/detail.html]: It should also receive a `page_parameter_name` so we can have multiple pagination types, on the same page.

File: ro_help/hub/views.py [hunk 6]
```diff
@@ -75,7 +75,42 @@ def get_context_data(self, **kwargs):
...
+        ngo = kwargs.get("ngo", context.get("ngo"))
+        if not ngo:
+            return context
+        page = self.request.GET.get("page")
+
+        # Donations paginator.
+        ngo_donations = ngo.get_funders()
+        donations_paginator = paginator.Paginator(ngo_donations.order_by("id"), self.paginated_by)
+        try:
+            donations_page_obj = donations_paginator.page(page)
```
Comment (vtemian) [inline ro_help/hub/views.py]: Should take into account `donation_page`.

File: ro_help/hub/views.py [hunk 7]
```diff
@@ -75,7 +75,42 @@ def get_context_data(self, **kwargs):
...
+        except (paginator.PageNotAnInteger, paginator.EmptyPage):
+            donations_page_obj = donations_paginator.page(1)
+        context[f"donations_page_obj"] = donations_page_obj
+
+        # Report items paginator.
+        ngo_report_items = ngo.report_items.all()
+        report_items_paginator = paginator.Paginator(ngo_report_items.order_by("id"), self.paginated_by)
+        try:
+            report_items_page_obj = report_items_paginator.page(page)
```
Comment (vtemian) [inline ro_help/hub/views.py]: Should take into account `report_item_page`.

**Comment by adriangabrieloros** [comment]
@costibleotu All the accordions need to be kept open, so I propose another specifically for that. 
@vtemian we have multiple types of pagination now.

File: ro_help/hub/views.py [hunk 8]
```diff
@@ -53,16 +52,15 @@ def get_context_data(self, **kwargs):
...
-
         needs = ngo.needs.filter(resolved_on=None)
         if "need" in kwargs:
             needs = needs.exclude(pk=kwargs["need"].pk)
             context["current_need"] = kwargs["need"]
 
         for kind in KIND.to_list():
             kind_needs = needs.filter(kind=kind)
-            needs_paginator = paginator.Paginator(kind_needs, self.paginated_by)
+            needs_paginator = paginator.Paginator(kind_needs, 3)
```
Comment (vtemian) [inline ro_help/hub/views.py]: Let's keep `paginated_by ` as a class attribute. Maybe also we can move it as a constant.

File: ro_help/hub/templates/shared/pagination.html [hunk 9]
```diff
@@ -4,10 +4,10 @@
 <nav class="pagination is-centered" role="navigation" aria-label="pagination">
     <span class="step-links">
         {% if page_obj.has_previous %}
-        <a class="pagination-previous" href="{% spurl base='{{ request.get_full_path }}' set_query='page=1' %}">&laquo;
+        <a class="pagination-previous" href="{% spurl base='{{ request.get_full_path }}' set_query='{{ page_parameter_name }}=1' %}">&laquo;
             {% trans "first" %}</a>
         <a class="pagination-previous"
-            href="{% spurl base='{{ request.get_full_path }}' set_query='page={{ page_obj.previous_page_number }}' %}">{% trans "previous" %}</a>
+            href="{% spurl base='{{ request.get_full_path }}' set_query='{{ page_parameter_name }}={{ page_obj.previous_page_number }}' %}">{% trans "previous" %}</a>
```
Comment (vtemian) [inline ro_help/hub/templates/shared/pagination.html]: `page_parameter_name` can be by default `page`.

**Pull request merged**